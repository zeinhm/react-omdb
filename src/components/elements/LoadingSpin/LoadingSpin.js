import React from 'react';
import PropTypes from 'prop-types';
import './styles.scoped.css';

export default function LoadingBars(props) {
  const { className, size } = props;
  const primaryColor = '#3498db';
  const remSize = `${size}rem`;
  const border = `${size/9}rem solid #f3f3f3`;
  const borderBottom = `${size/9}rem solid ${primaryColor}`;
  const borderRight = `${size/9}rem solid ${primaryColor}`;
  const borderTop = `${size/9}rem solid ${primaryColor}`;
  const customClass = ['loading-spinner', className].filter(Boolean).join(' ');
  return (
    <section className={customClass}>
      <div
        style={{
          border: border,
          borderBottom: borderBottom,
          borderRight: borderRight,
          borderTop: borderTop,
          height: remSize,
          width: remSize
        }}
      />
    </section>
  );
}

LoadingBars.defaultProps = {
  className: '',
  size: 4,
};

LoadingBars.propTypes = {
  className: PropTypes.string,
  size: PropTypes.number,
};
