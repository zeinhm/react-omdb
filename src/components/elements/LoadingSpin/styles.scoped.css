.loading-spinner {
  height: 100%; 
  left: 0;
  top: 0;
  width: 100%;
}

.loading-spinner > div {
  animation: spin 2s linear infinite;
  border: 16px solid #f3f3f3;
  border-bottom: 16px solid #3498db;
  border-radius: 50%;
  border-right: 16px solid #3498db;
  border-top: 16px solid #3498db;
  height: 150px;
  padding: .5rem;
  position: absolute;
  right: 50%;
  top: 30vh;
  transform: translate(-50%);
  width: 150px;
  -webkit-animation: spin 2s linear infinite;
}

@-webkit-keyframes spin {
  0% { -webkit-transform: rotate(0deg); }
  100% { -webkit-transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}