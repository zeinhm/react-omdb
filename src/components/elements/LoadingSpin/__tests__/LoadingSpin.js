import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoadingSpin from '../LoadingSpin';

describe('src/components/elements/LoadingSpin', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<LoadingSpin />);
    expect(tree).toMatchSnapshot();
  });
});
