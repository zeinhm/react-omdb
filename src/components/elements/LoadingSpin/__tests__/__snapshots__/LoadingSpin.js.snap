// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/LoadingSpin render 1`] = `
<section
  className="loading-spinner"
>
  <div
    style={
      Object {
        "border": "0.4444444444444444rem solid #f3f3f3",
        "borderBottom": "0.4444444444444444rem solid #3498db",
        "borderRight": "0.4444444444444444rem solid #3498db",
        "borderTop": "0.4444444444444444rem solid #3498db",
        "height": "4rem",
        "width": "4rem",
      }
    }
  />
</section>
`;
